pkg/api/getCoinData.go

```diff
@@ -168,41 +173,30 @@ func GetCoinHistory(ctx context.Context, id string, intervalChannel chan string,
 			break
 		}
 
-		url := fmt.Sprintf("https://api.coincap.io/v2/assets/%s/history?interval=%s", id, i)
-		data := CoinHistory{}
+		var intervalDuration float64 = intervalToDuration[i]
+		days := "1"
 
-		// Create Request
-		req, err := http.NewRequestWithContext(ctx, method, url, nil)
-		if err != nil {
-			finalErr = err
-			return
+		if intervalDuration <= 1800 {
+			days = "1"
+		} else if intervalDuration <= 43200 {
+			days = "90"
+		} else {
+			days = "360"
 		}
 
-		// Send Request
-		res, err := client.Do(req)
-		if err != nil {
-			finalErr = err
-			return
-		}
-		defer res.Body.Close()
+		data, _ := geckoClient.CoinsIDMarketChart(id, "usd", days)
```

Comment:
@temanisparsh Rather than ignore this error, you could make use of the `finalErr` and the deferred function call which returns error through the `errChan`. You can do this by simply checking if the error is nil, and if not, assign it to `finalErr` and return. You can look at how it's being done in other places before a `return` to get an idea.